### PR TITLE
Non-functional Requirements DG Update

### PR DESCRIPTION
**Pull request by daniellau88**
- Update Non functional requirements
- Fix minor bugs
- Add personal profile

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -727,6 +727,14 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Non-Functional Requirements
 
+1. Should work on any mainstream OS as long as it has Java 11 or above installed.
+
+1. Should be able to recover latest state of CHIM (before the command that causes the crash was executed) if the application crashes.
+
+1. Should be able to hold up to 1000 customers without a noticeable sluggishness in performance for typical usage.
+
+1. General queries (especially filter queries) should be completed within 10 seconds.
```
Comment (aaronsms) [inline docs/DeveloperGuide.md]: Is 10 seconds too much of a time, do we need that much of the time?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -727,6 +727,14 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Non-Functional Requirements
 
+1. Should work on any mainstream OS as long as it has Java 11 or above installed.
+
+1. Should be able to recover latest state of CHIM (before the command that causes the crash was executed) if the application crashes.
```
Comment (laurenlhy) [inline docs/DeveloperGuide.md:738]: numbering is all 1?

Comment (daniellau88) [reply docs/DeveloperGuide.md:738]: If you view in markdown, the numbering will be autonumbered correctly

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (daniellau88) [reply docs/DeveloperGuide.md]: Updated to 5 seconds

File: docs/UserGuide.md [hunk 3]
```diff
@@ -157,7 +157,7 @@ Example:
 Adds a cheese to CHIM.
 
 Format: `addcheese t/CHEESE_TYPE q/QUANTITY [d/MANUFACTURE_DATE] [e/EXPIRY_DATE]`
-* The specified `QUANTITY` must be a positive integer.
+* The specified `QUANTITY` must be a positive integer and less than 1000.
```
Comment (laurenlhy) [inline docs/UserGuide.md]: If upper limit is 1000, this should be "less than or equal to" 1000?

Comment (daniellau88) [reply docs/UserGuide.md]: Modified in latest commit

**Review by laurenlhy — APPROVED** [review]
LGTM

**Pull request merged**